### PR TITLE
Document VReplication Tablet Selection

### PR DESCRIPTION
**Pull request by mattlord**
And correct/unify related flag docs.

Additional details can be seen [here](https://gist.github.com/mattlord/794fcfd7e9a5eb2b45839fbe55e4f10e).

Signed-off-by: Matt Lord <mattalord@gmail.com>

### CODE REVIEW

**Review by shlomi-noach — COMMENTED** [review]
@mattlord thanks for documenting this! Would you be able to add docs for a recent change (undocumented yet)?

It's now possible to set an `in_order:` prefix instruction, e.g. `in_order:replica,rdonly`. With this  instruction, and in this example, vitess will first look for a random `replica`, and if none is available, a random `rdonly`.

The common use case will be `in_order:replica,primary` where we want the source stream to run off a `replica` tablet, and fallback to the `primary` tablet.

**Comment by mattlord** [comment]
@shlomi-noach added here: https://github.com/vitessio/website/pull/816/commits/82cf39f96587633b84557450753bc205eb2afd93

Thank you for that! I wasn't even aware of that new option... I like it. I'll also reach out to @deepthi about potentially making "in_order:REPLICA,PRIMARY" the default in 12.0. -- IMO that would be ideal.

File: content/en/docs/reference/vreplication/tablet_selection.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+weight: 300
+---
+
+### Introduction
+
+For [VReplication streams](../../../concepts/vstream/), we must choose a tablet to serve the role of source (vstreamer) and target (vapplier) in the replication stream and this is done automatically.
+
+To select the tablets we get a set of viable -- healthy and serving -- candidates for the source and target of the stream:
+  * **Source**: a random tablet is selected from the viable candidates of the specified types (see [tablet types](./#tablet-types))
+  * **Target**: a viable primary tablet is chosen as we need to do writes that are then replicated within the target shard
```
Comment (hallaroo) [inline content/en/docs/reference/vreplication/tablet_selection.md]: Just a possible grammar change:

* **Target**: a viable primary tablet is chosen, as we need to do writes that are then replicated within the target shard

File: content/en/docs/reference/vreplication/tablet_selection.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+### Cell considerations
+
+VReplication will only look for tablet pairings within the same cell. If you want to have cross-cell streams then you will need to [create a CellAlias](https://vitess.io/docs/reference/programs/vtctl/cell-aliases/) that contains the list of potential cells and specify that using the `-cell` flag in your VReplication workflow commands.
+
+### Tablet types
+
+The server side default which determines the candiate types made available for potential selection in a stream is set using the [vttablet's `-vreplication_tablet_type` flag](../flags/#vreplication_retry_delay) (the target tablet will use this in finding the viable source tablet candidates).
```
Comment (hallaroo) [inline content/en/docs/reference/vreplication/tablet_selection.md]: Just a possible grammar change:

The server side default which determines the candiate types made available for potential selection in a stream is set using the [vttablet's `-vreplication_tablet_type` flag](../flags/#vreplication_retry_delay). The target tablet will use this when finding the viable source tablet candidates.

**Review by hallaroo — APPROVED** [review]
Added a couple suggestions, but approving as it looks like the other comments we resolved and mine are grammar oriented

File: content/en/docs/reference/vreplication/tablet_selection.md (hunk 1 above)
Comment (mattlord) [reply content/en/docs/reference/vreplication/tablet_selection.md]: Sure, done.

File: content/en/docs/reference/vreplication/tablet_selection.md (hunk 2 above)
Comment (mattlord) [reply content/en/docs/reference/vreplication/tablet_selection.md]: Done

**Pull request merged**